azure-pipelines.yml

```diff
@@ -252,8 +312,14 @@ jobs:
         ISPARENT: True
   steps:
   - bash: |
+      uuid="$(tr -dc '[:alnum:]' </dev/urandom| head -c 16)"
       docker build -t ${DOCKER}:${VERSION}-${VARIANT} --build-arg MAKEFLAGS="-j$(($(grep -c ^processor /proc/cpuinfo) + 1))" docker-images/${VERSION}/${VARIANT}
-      docker run --rm ${DOCKER}:${VERSION}-${VARIANT} -buildconf
+      # Create a uniquely tagged version of the container to avoid pulling a version that was already on docker hub
```

Comment:
not sold on that, if you have a local image with the exact tag to run, docker won't pull from the registry

Reply:
But when the build actually fails you _don't_ have a local image with that tag to run, which is why it pulls from the registry when the build fails.

If you don't use an image name that exists in the registry (ie. something that is unique and local) then you can validate that it runs with a normal docker run without risking it pulling the image from the internet because the same tag already exists.

There's a proposal to make it possible to use a `--pull=never` flag here: https://github.com/docker/cli/issues/2253 but it's probably not going to make it into a distro version of docker anytime soon.

Reply:
So this is the sort of error case that doesn't get picked up if you don't use a unique name for the container to test:
https://dev.azure.com/video-tools/ffmpeg/_build/results?buildId=249&view=logs&j=6c6ac11e-9aaf-54db-7582-26c7d72f5185&t=c27d90b2-ebe3-5e65-dd76-20de7046977f&l=13782

If you look at the lines below that, it simply ends up just pulling the image from dockerhub and is obviously successfuly, because the version that was published obviously did build successfully.